### PR TITLE
support for logo and resource migration

### PR DESCRIPTION
**Pull request by perceptron-XYZ**
Logo images are now automatically loaded when doing "ckanapi load groups/organizations".

Resources can be loaded when --resources option is specified. If not, then the original links will be mapped to new CKAN instance.

### CODE REVIEW

**Comment by perceptron-XYZ** [comment]
Logos for groups and organizations from external links won't be uploaded to server now, new site will just copy their old link.

**Comment by perceptron-XYZ** [comment]
There goes all the fixes on options and styles

File: ckanapi/cli/load.py [hunk 1]
```diff
@@ -249,3 +264,37 @@ def _copy_from_existing_for_update(obj, existing, thing):
...
+                resource['url_type'] = ''           # hack url_type so that url can be modified
+                resource['package_id'] = obj['name']
+            ckan.action.resource_update(**resource)
+
+
+def _upload_logo(ckan,obj):
+    for key in obj.keys():
+        if isinstance(obj[key],(dict,list)):
+            obj.pop(key)                            #dict/list objects can't be encoded
+    if 'http' in obj['image_url']:                  # logo is an external link
```
Comment (wardi) [inline ckanapi/cli/load.py]: obj['image_url'].startswith('http') would be a little better. Even better would be to check for a netloc returned from urlparse https://docs.python.org/2/library/urlparse.html

File: ckanapi/cli/main.py [hunk 2]
```diff
@@ -4,7 +4,13 @@
   ckanapi action ACTION_NAME
           [KEY=VALUE ... | -i | -I JSON_INPUT] [-j | -J]
           [[-c CONFIG] [-u USER] | -r SITE_URL [-a APIKEY] [-g]]
-  ckanapi load (datasets | groups | organizations | users | related)
+  ckanapi load (datasets)
```
Comment (wardi) [inline ckanapi/cli/main.py]: no need for () around "datasets" here, and why not move the [--upload-resources] up to the top line so it's more obvious what option is special for loading datasets. Same for load (groups | organizations)

**Comment by perceptron-XYZ** [comment]
Hi Ian, just fixes a few more things you pointed out.
For the image_url attribute of logo images, it either starts with http (for external links) or a string of digits ( for uploaded images, digits basically represent the date & time of upload), so I think to check for the existence of urlparse(image_url).scheme should be enough, for the former case will give your 'http', and the latter gives you ''.

**Comment by perceptron-XYZ** [comment]
good point. Just changed it.

**Comment by wardi** [comment]
nicely done :+1:

**Pull request merged**